apps/web/src/components/pages/auth/logout.tsx

```diff
@@ -32,23 +32,30 @@ export function Logout({ oauth }: { oauth?: boolean }) {
     checkInstance({ redirect: false })
 
     const cookieSession = AuthSessionCookie.parse()
-    const isSSO = cookieSession?.authentication_methods
-      ? cookieSession.authentication_methods[0].method === 'oidc'
-      : false
+    const ssoProvider = cookieSession?.authentication_methods
+      ? cookieSession.authentication_methods[0].provider
+      : undefined
 
-    const redirection = isSSO
-      ? 'https://aai.demo.meinbildungsraum.de/realms/nbp-aai/protocol/openid-connect/logout'
-      : filterUnwantedRedirection({
-          desiredPath: sessionStorage.getItem('previousPathname'),
-          unwantedPaths: [settingsUrl, loginUrl, registrationUrl],
-        })
+    const redirection =
```

Comment:
Nitpick: Can we make this a key value map or something cleaner than the nested ternary's?

Reply:
I've just kept the style convention of the repo, but sure you can refactor it. I think a function would be the most readable refactoring.